### DISCUSSION TITLE
Problema despliegue prometheus

### DISCUSSION THREAD

**Discussion by Santiago21112001** [Arquisoft/faq#73] [Q&A] [answered]
Buenas, somos el equipo wiq_es0d. Al desplegar los dockers en la máquina de Azure, el contenedor de prometheus sufre un error y se apaga. Este es el error:
_ts=2024-04-27T12:02:08.637Z caller=main.go:521 level=error msg="Error loading config (--config.file=/etc/prometheus/prometheus.yml)" file=/etc/prometheus/prometheus.yml err="open /etc/prometheus/prometheus.yml: no such file or directory"_

Miré por qué pasaba y vi esto en el volumes de prometheus de docker-compose.yml:
_./gatewayservice/monitoring/prometheus:/etc/prometheus_

Por lo que yo entiendo entonces se necesita en la máquina virtual todo el contenido de gatewayservice/monitoring, igual que se necesita el docker-compose-yml o el .env.

![imagen](https://github.com/Arquisoft/faq/assets/79706368/e8936543-4802-41be-a5d0-d47d865bd16c)
![imagen](https://github.com/Arquisoft/faq/assets/79706368/02aa931c-219c-4a2c-acb5-149e4d8831fe)

¿Cómo se puede arreglar? En local funciona sin problemas

**Comment by Santiago21112001** [comment, marked as answer]
Lo arreglé poniendo todo el contenido de gatewayservice/monitoring del repositorio github en el directorio gatewayservice/monitoring de mi usuario de la maquina de azure. Luego reinicié los contenedores con:
          _docker compose --profile prod down
          docker compose --profile prod up -d --pull always_
Ahora funciona grafana y prometheus en la máquina remota y se pueden ver los datos de monitorización.
El inconveniente de esta solución es que si se cambia algún fichero de configuración de los que están en gatewayservice/monitoring, hay que actualizarlos manualmente también en la máquina remota.

**Reply by pglez82** [reply]
Lo correcto sería modificar el actions 'deploy' para que copie o descargue ese contenido en la máquina virtual antes de lanzar el docker-compose. Así si haces algún cambio a los dashboards o la configuración de la monitorización no tienes que acordarte de volver a bajarlo, con redesplegar sería suficiente.
